### PR TITLE
Cleanup deprecated constructor use

### PR DESCRIPTION
**Pull request by BradWalker**
Cleaning some old code to make it more current..

- removed use of old deprecated constructor use
- use autobox/unbox for encapsulation
- change chain-of-if statement to switch w/ string
- some code cleanup

---
**^Add meaningful description above**

By opening a pull request you confirm that, unless explicitly stated otherwise, the changes -

 - are all your own work, and you have the right to contribute them.
 - are contributed solely under the terms and conditions of the Apache License 2.0 (see section 5 of the license for more information).

Please make sure (eg. `git log`) that all commits have a valid name and email address for you in the Author field.

If you're a first time contributor, see the Contributing guidelines for more information.

### CODE REVIEW

File: profiler/lib.profiler/src/org/netbeans/lib/profiler/heap/ClassDumpSegment.java [hunk 1]
```diff
@@ -235,55 +233,118 @@ void addInstanceSize(ClassDump cls, int tag, long instanceOffset) {
...
+                    java_lang_Class = jcls;
+                    break;
+                    
+                case "java.lang.Object":
+                    // NOI18N
+                    java_lang_Object = jcls;
+                    break;
+                    
+                default:
+                    break;
```
Comment (mbien) [inline profiler/lib.profiler/src/org/netbeans/lib/profiler/heap/ClassDumpSegment.java:347]: the intention is good but i think this makes it more difficult to read.

try to format it like:
```java
               case "[Z":  typeObj = HprofHeap.BOOLEAN;  break; // NOI18N
```
Its non-standard but would be compact and easier to read in this particular case. The "// NOI18N" hint must be on the same line as the String literal. 

Formatting it like this is also really close to switch expressions of later JDKs. Updating to them would only require the "->" and removal of "break".

But most importantly: we have to be careful with null here. What if vmName is null? The switch will throw an exception, the if-else would do fine.

File: profiler/lib.profiler/src/org/netbeans/lib/profiler/server/ProfilerRuntimeCPU.java [hunk 2]
```diff
@@ -338,14 +338,29 @@ static void writeServletDoMethod(ThreadInfo ti, String method, String servletPat
...
-            methodId = 1;
-        } else if ("POST".equals(method)) { // NOI18N
-            methodId = 2;
-        } else if ("PUT".equals(method)) { // NOI18N
-            methodId = 3;
-        } else if ("DELETE".equals(method)) { // NOI18N
-            methodId = 4;
+        if (null != method) switch (method) {
+            case "GET":
+                // NOI18N
```
Comment (mbien) [inline profiler/lib.profiler/src/org/netbeans/lib/profiler/server/ProfilerRuntimeCPU.java:343]: same here: // NOI18N must be on the same line behind case "foo"

**Comment by mbien** [comment]
@BradWalker  please don't merge PRs without giving others a chance to review - even if it seems trivial.

**Comment by BradWalker** [comment]
@mbien , sorry about that.. I got a little bit aggressive with this.. My sincere apologies.. I'll get it reverted..

**Pull request merged**